ansible/templates/nginx.conf

```diff
@@ -70,12 +70,21 @@ server {
         proxy_cache_min_uses    5;
     }    
 
-    location /pictshare/ {
+    # Redirect pictshare images to pictrs
+    rewrite ^(/pictshare/.*)$ /pictrs/image/$1;
+
+    # pict-rs images
+    location /pictrs/image/ {
       proxy_pass http://0.0.0.0:8537/;
       proxy_set_header X-Real-IP $remote_addr;
       proxy_set_header Host $host;
       proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
 
+      # Block the import
+      location /pictrs/import {
+        return 403;
+      }
```

Comment:
This is not working, `nginx -t` says `location "/pictrs/import" is outside location "/pictrs/image/" in /etc/nginx/sites-enabled/lemmy.conf:84`.

Reply:
Your routing is generally wrong, currently images are served under `/pictrs/image/image/008jwy.jpg`.

Reply:
Here is a working config:
```
# pict-rs images
location /pictrs {
    location /pictrs/image {
        proxy_pass http://0.0.0.0:8537/image/;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header Host $host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    }
    # Block the import
    return 403;
}
```

Reply:
Heres a working redirect (this should be more efficient than `rewrite`).
```
location ~ /pictshare/(.*)$ {
    return 301 /pictrs/image/$1;
}
```